thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/DetectionHealth.java

```diff
@@ -0,0 +1,379 @@
+     * @return the builder
+     */
+    public Builder addOverallHealth() {
+      this.provideOverallHealth = true;
+      return this;
+    }
+
+    /**
+     * Build the health status object
+     * @return the health status object
+     */
+    public DetectionHealth build() {
+      DetectionHealth health = new DetectionHealth();
+      if (this.evaluationDAO != null) {
+        health.regressionStatus = buildRegressionStatus();
+      }
+      if (this.anomalyDAO != null) {
+        health.anomalyCoverageStatus = buildAnomalyCoverageStatus();
+      }
+      if (this.taskDAO != null) {
+        health.detectionTaskStatus = buildTaskStatus();
+      }
+      if (this.provideOverallHealth) {
+        health.overallHealth = classifyOverallHealth(health);
+      }
+      return health;
+    }
+
+    private RegressionStatus buildRegressionStatus() {
+      // fetch evaluations
+      List<EvaluationDTO> evaluations = this.evaluationDAO.findByPredicate(
+          Predicate.AND(Predicate.LT(COL_NAME_START_TIME, endTime), Predicate.GT(COL_NAME_END_TIME, startTime),
+              Predicate.EQ(COL_NAME_DETECTION_CONFIG_ID, detectionConfigId)));
+
+      // calculate average mapes for each detector
+      Map<String, Double> detectorMapes = evaluations.stream()
+          .filter(eval -> Objects.nonNull(eval.getMape()))
+          .collect(Collectors.groupingBy(EvaluationBean::getDetectorName,
+              Collectors.averagingDouble(EvaluationBean::getMape)));
+
+      // construct regression status
+      RegressionStatus status = new RegressionStatus();
+      status.detectorMapes = detectorMapes;
+      status.detectorHealthStatus = detectorMapes.entrySet()
+          .stream()
+          .collect(Collectors.toMap(Map.Entry::getKey, e -> classifyMapeHealth(e.getValue())));
+      status.healthStatus = classifyOverallRegressionStatus(status.detectorHealthStatus);
+      return status;
+    }
+
+    private static HealthStatus classifyMapeHealth(double mape) {
+      if (mape < 0.2) {
+        return HealthStatus.GOOD;
+      }
+      if (mape < 0.5) {
+        return HealthStatus.MODERATE;
+      }
+      return HealthStatus.BAD;
+    }
+
+    /**
+     * Classify the regression status of the detection config based on the health status for each detector
+     * @param detectorHealthStatus the health status for each detector
+     * @return the overall regression status
+     */
+    private static HealthStatus classifyOverallRegressionStatus(Map<String, HealthStatus> detectorHealthStatus) {
+      if (detectorHealthStatus.values().contains(HealthStatus.GOOD)) {
+        return HealthStatus.GOOD;
+      }
+      if (detectorHealthStatus.values().contains(HealthStatus.MODERATE)) {
+        return HealthStatus.MODERATE;
+      }
+      return HealthStatus.BAD;
+    }
+
+    private AnomalyCoverageStatus buildAnomalyCoverageStatus() {
+      List<MergedAnomalyResultDTO> anomalies = this.anomalyDAO.findByPredicate(
```

Comment:
You could enclose these builder methods in try-catch and throw a meaningful message.

Reply:
Added the try-catch in the outer level